### ISSUE TITLE
Window resizing broken in Ubuntu 20.04 with Wayland

### ISSUE THREAD

**Issue by uusijani** [wezterm/wezterm#2563]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome

### WezTerm version

20220927-071112-9be05951

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I'm running a standard Ubuntu 20.04 Gnome desktop, but with Wayland. Resizing of wezterm window is broken on this setup: selecting "maximize" from the title bar context menu or dragging the window against the top or the sides of the screen doesn't maximize/tile the window, but any resizing that does happen results in any text entered in the terminal thereafter to be get garbled. In this screenshot I've first entered `wezterm --version` before resizing, then dragged the window against the top, then entered  `wezterm --version` again:

![Screenshot from 2022-09-28 13-43-02](https://user-images.githubusercontent.com/4244772/192759722-c3fa1c27-e651-4c17-a1eb-adb7002686f1.png)

### To Reproduce

1. Install your bog-standard Ubuntu 20.04 desktop in a VM and boot it up.
2. On the GDM login screen select "Ubuntu on Wayland" from the bottom right menu, then log in.
3. Install wezterm.
4. Start wezterm.
5. Right-click wezterm's title bar and select "Maximize".
6. Type something

### Configuration

no config

### Expected Behavior

After step 5, I expect the window to be maximized (i.e. to fill the desktop). After step 6, I expect to see what I typed.

### Logs

no logs available

### Anything else?

The issue is not present in an X11 session. Ubuntu 20.04 defaults to X11, but I've used it with Wayland exclusively since the release without issues. More specifically, I can't remember seeing resizing issues similar to this with any other application.

I also have a laptop running Ubuntu 22.04 and Wayland, and there all resizing of the wezterm window works just as I'd expect (i.e. just as in any other application).

**Comment by wez** [comment]
> 5\. Right-click wezterm's title bar and select "Maximize".

Please tell me more about your window environment; Gnome/Mutter don't support Client Side Decoration (CSD) for Wayland clients, so wezterm draws its own limited decorations.  Those don't support right clicking or context menus, so I'm not sure how that maximize menu you described shows up.

I would also like to see the logs from the debug overlay: press CTRL-SHIFT-L and copy and paste the lines of text you see there into this issue.

**Comment by tkinz27** [comment]
So I saw weird behavior on arch (using latest gnome) and ubuntu 22 when I set `window_decorations = 'NONE'`. Whenever wezterm is maximized to half my screen and I switch applications, the terminal "resizes" to the previous (i think) size. I put "resizes" in quotes cause I have to unmaximize the terminal and remaximize it to get it back to the the half screen size. 

Not sure if its related but seems fairly similar. 

```
❯ dpkg -l | grep mutter
ii  gir1.2-mutter-10:amd64                         42.5-0ubuntu1                                                             amd64        GObject introspection data for Mutter
ii  libmutter-10-0:amd64                           42.5-0ubuntu1                                                             amd64        window manager library from the Mutter window manager
ii  mutter                                         42.5-0ubuntu1                                                             amd64        Example window manager using GNOME's window manager library
ii  mutter-common                                  42.5-0ubuntu1                                                             all          shared files for the Mutter window manager
```

**Comment by andradei** [comment]
Having this issue on
- KDE Plasma 5.27.9 (Wayland)
- Wezterm 20230712-072601-f4abf8fd
- EndeavourOS (Arch)

**Comment by cpieloth** [comment]
I can not re-produce it on my system:

```bash
user@host:~$ echo $XDG_SESSION_TYPE
wayland
user@host:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 25.10
Release:        25.10
Codename:       questing
user@host:~$ wezterm --version
wezterm 20240203-110809-5046fc22
```

The mentioned version of wezterm is pretty old and Ubuntu 20.04 is out-of-support (except for ESM).

@uusijani @andradei Can you re-test it with the nightly, please?

**Comment by uusijani** [comment]
Yeah, seems to be fixed: I tested in an Ubuntu 24.04 vm with WezTerm 20240203-110809-5046fc22 and couldn't reproduce the issue. 

(I first tried the nightly, which from the repository was 20260117-154428-05343b38, but that one was missing all window decorations entirely, and so couldn't be resized at all.)

I'll close the issue and unsubscribe, as I'm not using WezTerm currently myself. Others can reopen it if still affected.
